### PR TITLE
Add custom function validation feature

### PR DESCRIPTION
**Pull request by jhomarolo**
Add custom function validation feature about #23 , special thanks to @vortxcodasquieves to huge help with this contrib.

## Proposed Changes

1. Add custom function validation feature 
2. Add tests for single validation
3. Add testes for multiple validations

### Author/Contributor
- [X] If documentation is needed for this change, has that been included in this pull request
- [x] Remember to check if code coverage decrease, if so, please implement the necessary tests

### Reviewing Maintainer
- [x] Label as `breaking` if this is a large fundamental change
- [x] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

### CODE REVIEW

**Comment by jhomarolo** [comment]
I already did some tests on gotu with this functionality, I even wrote some unit tests and it worked well.

After the merge of this pr and npm update, I will make a new pr on gotu with the updates

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,3 +1,9 @@
 {
-  "python.pythonPath": "C:\\Python38\\python.exe"
+  "python.pythonPath": "C:\\Python38\\python.exe",
```
Comment (maikvortx) [inline .vscode/settings.json:2]: Should this configuration be updated?

Comment (jhomarolo) [reply .vscode/settings.json]: done

**Review by maikvortx — APPROVED** [review]

**Pull request merged**